src/ProfileSync.ts

```diff
@@ -90,9 +101,14 @@ export class ProfileSync {
                 matrixUser.set("avatar_url", remoteProfileSet.avatar_uri);
             } catch (e) {
                 log.error("Failed to update avatar_url for user:", e);
+                errors.push(e);
             }
         }
         await this.store.setMatrixUser(matrixUser);
+
+        if (errors.length) {
+            throw new AggregateError(errors, "Failed to fully update profile for user");
```

Comment:
Is this caught anywhere? Else won't it crash in the same way?

Reply:
Yes, because every call to `updateProfile` is now protected by a try-catch.

Some existing callers (namely `MatrixRoomHandler#handleRemoteUserState`) expect to catch failures on profile updates, so catching all such errors here would break that.